lib/compiler/libc.zig

```diff
@@ -113,7 +113,8 @@ pub fn main() !void {
         };
         defer libc.deinit(gpa);
     } else {
-        if (!target_query.isNative()) {
+        // macos can detect LibC for different SDK-based targets
+        if (!target_query.isNative() and (@import("builtin").os.tag != .macos)) {
```

Comment:
I don't think `(` `)` are necessary here, but  this is just a minor nit. This entire thing seems like a hack to me. I think we should somehow incorporate this into `target_query` by having something like `target_query.canDetectLibC`. @andrewrk thoughts?

Reply:
Going that route should also benefit android `-target aarch64-linux-android`, allowing detection via env `CC` and `cc -c -E -Wp,-v -xc /dev/null`.

Reply:
I went ahead and implemented `target_query.canDetectLibC`.